### DISCUSSION TITLE
Adding TextEdit padding support

### DISCUSSION THREAD

**Discussion by ponchofiesta** [emilk/egui#5235] [Q&A]
Hi,
I'm trying to implement padding support for TextEdit widget. But actually the size of the widget is not big enough.

Sample code:
```
impl eframe::App for MyApp {
    fn update(&mut self, ctx: &egui::Context, _frame: &mut eframe::Frame) {
        egui::CentralPanel::default().show(ctx, |ui| {
            ui.style_mut().spacing.textedit_padding = vec2(8.0, 8.0);
            ui.text_edit_singleline(&mut self.text);
        });
    }
}
```

![image](https://github.com/user-attachments/assets/d52f67e5-e844-4c83-8051-d00d02acae19)

The changes to egui are in my fork:
https://github.com/ponchofiesta/egui/compare/master...ponchofiesta:egui:feature/textedit-padding

I assumed this should enlarge the widget to fit text into it. But actually the text gets cut off. But the padding looks Ok so far. Only the text area needs to be bigger.

Could some one give me a hint how to fix this to make the widget big enough to make it work?

**Comment by YgorSouza** [comment]
[TextEdit::margin](https://docs.rs/egui/0.29.1/egui/widgets/text_edit/struct.TextEdit.html#method.margin) already does that.

![image](https://github.com/user-attachments/assets/cf1217eb-3f50-4872-b770-330ebc571238)

**Reply by ponchofiesta** [reply]
This is strange. For a button Margin sets the outer space and padding the inner space (as you would expect coming from CSS). And for TextEdit margin sets the padding? I didn't verify it but this sound not like it should be.

**Reply by ponchofiesta** [reply]
I verified it and margin does set the padding correctly.

**Reply by YgorSouza** [reply]
Egui's styling story has some inconsistencies like that. There are some open issues to improve it, such as #4019, but it's a work in progress.
